test/generic/Matrix-test.jl

```diff
@@ -1554,6 +1554,124 @@ end
    end
 end
 
+@testset "Generic.Mat.can_solve..." begin
+   for R in [ZZ, QQ]
+      for iter = 1:40
+         for dim = 0:5
+            r = rand(1:5)
+            n = rand(1:5)
+            c = rand(1:5)
+
+            let
+               S = MatrixSpace(R, n, r)
+               U = MatrixSpace(R, c, n)
+
+               X1 = rand(S, -20:20)
+               M = rand(U, -20:20)
+
+               B = M*X1
+               (flag, X) = can_solve_with_solution(M, M*X1)
+
+               @test can_solve(M, B)
+               @test flag && M*X == B
+            end
+
+            let
+               S = MatrixSpace(R, r, n)
+               U = MatrixSpace(R, n, c)
+
+               X1 = rand(S, -20:20)
+               M = rand(U, -20:20)
+
+               B = X1*M
+               (flag, X) = can_solve_with_solution(M, X1*M; side = :left)
+
+               @test can_solve(M, B; side = :left)
+               @test flag && X*M == B
+            end
+         end
+      end
+   end
+
+   R, x = PolynomialRing(QQ, "x")
+
+   for iter = 1:4
+      for dim = 0:5
+         r = rand(1:5)
+         n = rand(1:5)
+         c = rand(1:5)
+
+         let
+            S = MatrixSpace(R, n, r)
+            U = MatrixSpace(R, c, n)
+
+            X1 = rand(S, 1:2, -10:10)
+            M = rand(U, 1:2, -10:10)
+
+            B = M*X1
+            (flag, X) = can_solve_with_solution(M, M*X1)
```

Comment:
Maybe you can test also `can_solve_with_solution` with `side = :right` ? And test that an error is thrown when `side` is not `:right` or `:left`.

Reply:
What do you suggest to test for `side`? Something nonsensical like `side=:aaa`, or a random symbol, or something else?

Reply:
Yes either `:aaa` or `Symbol(randstring())` will do. And why not with something else, like the string `"right"`.